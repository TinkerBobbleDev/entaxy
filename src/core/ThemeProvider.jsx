import React from 'react'
import PropTypes from 'prop-types'
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles'
import CssBaseline from '@material-ui/core/CssBaseline'
import blueGrey from '@material-ui/core/colors/blueGrey'
import '../styles/globals.css'

const theme = createMuiTheme({
  palette: {
    primary: { main: blueGrey[800] },
    secondary: { main: '#1B9CC4' },
    info: {
      text: '#00acc1',
      background: '#00acc11a'
    },
    danger: {
      text: '#721c24',
      background: '#f8d7da66'
    }

  },
  typography: {
    useNextVariants: true
  }
})

const ThemeProvider = ({ children }) => (
  <CssBaseline>
    <MuiThemeProvider theme={theme}>
      {children}
    </MuiThemeProvider>
  </CssBaseline>
)

ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired
}

export default ThemeProvider
