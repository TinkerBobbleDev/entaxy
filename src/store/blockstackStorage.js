/* eslint-disable no-console */
import { getFile, putFile, isUserSignedIn } from 'blockstack'

const filename = 'entaxy.json'

export const loadState = () => {
  if (isUserSignedIn()) {
    return getFile(filename)
      .then(data => JSON.parse(data))
      .catch((error) => {
        console.log(`Error loading file ${filename}:`, error)
        return {}
      })
  }
  return Promise.resolve({})
}

export const saveState = (state) => {
  if (isUserSignedIn()) {
    return putFile(filename, JSON.stringify(state))
  }
  return undefined
}
