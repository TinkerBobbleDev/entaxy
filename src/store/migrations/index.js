import migration0 from './0'
import migration1 from './1'
import migration2 from './2'
import migration3 from './3'

export default {
  0: migration0,
  1: migration1,
  2: migration2,
  3: migration3
}
