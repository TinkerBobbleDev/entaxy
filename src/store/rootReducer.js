import { combineReducers } from 'redux'
import userReducer from './user/reducer'
import settingsReducer from './settings/reducer'
import accountsReducer from './accounts/reducer'
import transactionsReducer from './transactions/reducer'
import exchangeRatesReducer from './exchangeRates/reducer'
import budgetReducer from './budget/reducer'

export default combineReducers({
  user: userReducer,
  settings: settingsReducer,
  accounts: accountsReducer,
  transactions: transactionsReducer,
  exchangeRates: exchangeRatesReducer,
  budget: budgetReducer
})
