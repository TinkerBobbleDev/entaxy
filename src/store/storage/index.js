/* eslint-disable import/no-cycle */
import { snakeCase } from 'lodash'
import * as blockstackStorage from './blockstackStorage'
import * as localStorage from './localStorage'
import { showOverlay, hideOverlay } from '../settings/actions'
import { initialState as settingsInitialState } from '../settings/reducer'
import { initialState as accountsInitialState } from '../accounts/reducer'
import { initialState as transactionsInitialState } from '../transactions/reducer'
import { initialState as exchangeRatesInitialState } from '../exchangeRates/reducer'
import { initialState as budgetInitialState } from '../budget/reducer'

export const loadItem = (itemName, newState) => (dispatch, getState) => {
  let payload = newState
  if (itemName === 'settings') {
    const { settings } = getState()
    if (settings !== null && settings !== undefined) {
      payload = { ...newState, overlayMessage: settings.overlayMessage }
    }
  }
  dispatch({
    type: `LOAD_${snakeCase(itemName).toUpperCase()}`,
    payload
  })
}

export const loadAllItems = (newState) => (dispatch) => {
  [
    'settings',
    'accounts',
    'transactions',
    'exchangeRates',
    'budget'
  ].map((itemName) => dispatch(loadItem(itemName, newState[itemName] || {})))
}


export const loadState = () => async (dispatch, getState) => {
  if (getState().user === undefined) return {}
  let newState
  switch (getState().user.isAuthenticatedWith) {
    case 'guest':
      dispatch(showOverlay('Loading data from localStorage...'))
      newState = await localStorage.loadState()
      break
    case 'blockstack':
      dispatch(showOverlay('Loading data from Blockstack...'))
      newState = await blockstackStorage.loadState()
      break
    // no default
  }
  if (newState === undefined) newState = { settings: settingsInitialState }
  dispatch(loadAllItems({
    ...newState,
    settings: {
      ...newState.settings,
      snackbarMessage: null
    }
  }))
  return dispatch(hideOverlay())
}

export const resetState = () => (dispatch, getState) => {
  const { settings } = getState()

  // Preserve the existing currency and locale
  dispatch(loadAllItems({
    settings: {
      ...settingsInitialState,
      currency: settings.currency,
      locale: settings.locale
    },
    accounts: accountsInitialState,
    transactions: transactionsInitialState,
    exchangeRates: exchangeRatesInitialState,
    budget: budgetInitialState
  }))
}

export const saveState = () => (_, getState) => {
  let { user } = getState()
  if (user === undefined) user = {}
  switch (user.isAuthenticatedWith) {
    case 'guest':
      return localStorage.saveState(getState())
    case 'blockstack':
      return blockstackStorage.saveState(getState())
    default:
      return undefined
  }
}
