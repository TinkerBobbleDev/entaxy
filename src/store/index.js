import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunkMiddleware from 'redux-thunk'
import rootReducer from './rootReducer'

const middlewares = [thunkMiddleware]

if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger')
  middlewares.push(logger)
}

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middlewares))
)

if (module.hot) {
  module.hot.accept(rootReducer, () => {
    store.replaceReducer(rootReducer)
  })
}

export default store
