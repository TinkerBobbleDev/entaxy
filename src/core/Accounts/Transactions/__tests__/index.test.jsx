import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import store from '../../../../store'
import ThemeProvider from '../../../ThemeProvider'
import { TransactionsComponent } from '..'

jest.mock('../TransactionDialog', () => 'TransactionDialog')

describe('Transactions', () => {
  const account = {
    id: 1,
    name: 'Checking',
    institution: 'TD',
    currency: 'CAD',
    openingBalance: 0,
    openingBalanceDate: Date.parse('2019/01/01')
  }
  const transactions = [{
    id: 1,
    accountId: 1,
    amount: 10,
    createdAt: Date.now()
  }, {
    id: 2,
    accountId: 1,
    amount: -5,
    createdAt: Date.now() + 10
  }]
  const mochDeleteTransactions = jest.fn()
  const mochHandleSort = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot with no transactions', () => {
    const component = renderer.create((
      <Provider store={store}>
        <ThemeProvider>
          <BrowserRouter>
            <TransactionsComponent
              classes={{ }}
              account={account}
              deleteTransactions={mochDeleteTransactions}
              handleSort={mochHandleSort}
              transactions={[]}
            />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot with one transaction in a fiat account', () => {
    expect(account.type).toBeUndefined()
    const component = renderer.create((
      <Provider store={store}>
        <ThemeProvider>
          <BrowserRouter>
            <TransactionsComponent
              classes={{ }}
              account={account}
              deleteTransactions={mochDeleteTransactions}
              handleSort={mochHandleSort}
              transactions={transactions}
            />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  describe('Component methods', () => {
    describe('TransactionsDialog', () => {
      const wrapper = shallow((
        <TransactionsComponent
          classes={{ }}
          account={account}
          deleteTransactions={mochDeleteTransactions}
          handleSort={mochHandleSort}
          transactions={transactions}
        />
      ))
      const instance = wrapper.instance()

      it('opens new transaction dialog', () => {
        expect(wrapper.state('openTransactionDialog')).toBe(false)
        expect(wrapper.state('transaction')).toBe(null)
        instance.handleNew()
        expect(wrapper.state('openTransactionDialog')).toBe(true)
        expect(wrapper.state('transaction')).toBe(null)
      })

      it('closes transaction dialog', () => {
        expect(wrapper.state('openTransactionDialog')).toBe(true)
        expect(wrapper.state('transaction')).toBe(null)
        instance.handleCancel()
        expect(wrapper.state('openTransactionDialog')).toBe(false)
        expect(wrapper.state('transaction')).toEqual(null)
      })

      it('opens edit transaction dialog', () => {
        const transaction = { id: 1 }
        expect(wrapper.state('openTransactionDialog')).toBe(false)
        expect(wrapper.state('transaction')).toBe(null)
        instance.handleEdit(transaction)
        expect(wrapper.state('openTransactionDialog')).toBe(true)
        expect(wrapper.state('transaction')).toEqual(transaction)
      })
    })
  })
})
