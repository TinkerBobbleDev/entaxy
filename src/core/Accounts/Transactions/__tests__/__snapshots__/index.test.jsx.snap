// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions matches snapshot with no transactions 1`] = `
<div>
  <TransactionDialog
    account={
      Object {
        "currency": "CAD",
        "id": 1,
        "institution": "TD",
        "name": "Checking",
      }
    }
    onCancel={[Function]}
    open={false}
    transaction={null}
  />
  <div
    className="TransactionsTableComponent-tableWrapper-1 "
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular TableToolbar-root-16 MuiToolbar-gutters"
    >
      <div
        className="TableToolbar-title-20"
      >
        <div>
          <h6
            className="MuiTypography-root MuiTypography-h6"
          >
            TD - Checking
          </h6>
          <span
            className="MuiTypography-root MuiTypography-caption"
          >
            CAD
          </span>
        </div>
      </div>
      <div
        className="TableToolbar-spacer-18"
      />
      <div
        className="TableToolbar-actions-19"
      >
        <div
          className="withRouter(TransactionsToolbarComponent)-buttons-10"
        >
          <div
            className="withRouter(TransactionsToolbarComponent)-search-11"
          >
            <div
              className="withRouter(TransactionsToolbarComponent)-searchIcon-12"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              className="MuiInputBase-root withRouter(TransactionsToolbarComponent)-inputRoot-14"
              onClick={[Function]}
            >
              <input
                className="MuiInputBase-input withRouter(TransactionsToolbarComponent)-inputInput-15"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Search description or categories..."
                type="text"
                value=""
              />
            </div>
          </div>
          <a
            aria-describedby={null}
            aria-disabled={false}
            aria-label="Import transaction"
            className="MuiButtonBase-root MuiIconButton-root"
            href="/accounts/1/import"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
            title="Import transaction"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                className="withRouter(TransactionsToolbarComponent)-importButton-9"
                role="presentation"
                style={
                  Object {
                    "height": "1.5rem",
                    "width": "1.5rem",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M14,12L10,8V11H2V13H10V16M20,18V6C20,4.89 19.1,4 18,4H6C4.9,4 4,4.9 4,6V9H6V6H18V18H6V15H4V18C4,19.1 4.9,20 6,20H18C19.1,20 20,19.1 20,18Z"
                  style={Object {}}
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
          <button
            aria-describedby={null}
            aria-label="New transaction"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="New transaction"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    >
      <div
        aria-colcount={7}
        aria-rowcount={1}
        className="ReactVirtualized__Table"
        role="grid"
        style={Object {}}
      >
        <div
          className="ReactVirtualized__Table__headerRow TransactionsTableComponent-headerRow-2 TransactionsTableComponent-oddRow-5"
          role="row"
          style={
            Object {
              "height": 25,
              "overflow": "hidden",
              "paddingRight": 0,
              "width": 0,
            }
          }
        >
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 40px",
                "flex": "0 1 40px",
                "msFlex": "0 1 40px",
              }
            }
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-107 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <input
                    checked={false}
                    className="PrivateSwitchBase-input-110"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </span>
            </span>
          </div>
          <div
            aria-label="Date"
            aria-sort="descending"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Date"
            >
              Date
            </span>
            <svg
              className="ReactVirtualized__Table__sortableHeaderIcon ReactVirtualized__Table__sortableHeaderIcon--DESC"
              height={18}
              viewBox="0 0 24 24"
              width={18}
            >
              <path
                d="M7 10l5 5 5-5z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "1 1 200px",
                "flex": "1 1 200px",
                "msFlex": "1 1 200px",
              }
            }
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Description"
            >
              Description
            </span>
          </div>
          <div
            aria-label="Category"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 200px",
                "flex": "0 1 200px",
                "msFlex": "0 1 200px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Category"
            >
              Category
            </span>
          </div>
          <div
            aria-label="In"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 130px",
                "flex": "0 1 130px",
                "msFlex": "0 1 130px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="In"
            >
              In
            </span>
          </div>
          <div
            aria-label="Out"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 130px",
                "flex": "0 1 130px",
                "msFlex": "0 1 130px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Out"
            >
              Out
            </span>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 40px",
                "flex": "0 1 40px",
                "msFlex": "0 1 40px",
              }
            }
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title={null}
            />
          </div>
        </div>
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
          onScroll={[Function]}
          role="rowgroup"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": -25,
              "overflowX": "hidden",
              "overflowY": "auto",
              "position": "relative",
              "width": 0,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Transactions matches snapshot with one transaction in a fiat account 1`] = `
<div>
  <TransactionDialog
    account={
      Object {
        "currency": "CAD",
        "id": 1,
        "institution": "TD",
        "name": "Checking",
      }
    }
    onCancel={[Function]}
    open={false}
    transaction={null}
  />
  <div
    className="TransactionsTableComponent-tableWrapper-1 "
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular TableToolbar-root-16 MuiToolbar-gutters"
    >
      <div
        className="TableToolbar-title-20"
      >
        <div>
          <h6
            className="MuiTypography-root MuiTypography-h6"
          >
            TD - Checking
          </h6>
          <span
            className="MuiTypography-root MuiTypography-caption"
          >
            CAD
          </span>
        </div>
      </div>
      <div
        className="TableToolbar-spacer-18"
      />
      <div
        className="TableToolbar-actions-19"
      >
        <div
          className="withRouter(TransactionsToolbarComponent)-buttons-10"
        >
          <div
            className="withRouter(TransactionsToolbarComponent)-search-11"
          >
            <div
              className="withRouter(TransactionsToolbarComponent)-searchIcon-12"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              className="MuiInputBase-root withRouter(TransactionsToolbarComponent)-inputRoot-14"
              onClick={[Function]}
            >
              <input
                className="MuiInputBase-input withRouter(TransactionsToolbarComponent)-inputInput-15"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Search description or categories..."
                type="text"
                value=""
              />
            </div>
          </div>
          <a
            aria-describedby={null}
            aria-disabled={false}
            aria-label="Import transaction"
            className="MuiButtonBase-root MuiIconButton-root"
            href="/accounts/1/import"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
            title="Import transaction"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                className="withRouter(TransactionsToolbarComponent)-importButton-9"
                role="presentation"
                style={
                  Object {
                    "height": "1.5rem",
                    "width": "1.5rem",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M14,12L10,8V11H2V13H10V16M20,18V6C20,4.89 19.1,4 18,4H6C4.9,4 4,4.9 4,6V9H6V6H18V18H6V15H4V18C4,19.1 4.9,20 6,20H18C19.1,20 20,19.1 20,18Z"
                  style={Object {}}
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
          <button
            aria-describedby={null}
            aria-label="New transaction"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="New transaction"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    >
      <div
        aria-colcount={7}
        aria-rowcount={3}
        className="ReactVirtualized__Table"
        role="grid"
        style={Object {}}
      >
        <div
          className="ReactVirtualized__Table__headerRow TransactionsTableComponent-headerRow-2 TransactionsTableComponent-oddRow-5"
          role="row"
          style={
            Object {
              "height": 25,
              "overflow": "hidden",
              "paddingRight": 0,
              "width": 0,
            }
          }
        >
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 40px",
                "flex": "0 1 40px",
                "msFlex": "0 1 40px",
              }
            }
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-107 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <input
                    checked={false}
                    className="PrivateSwitchBase-input-110"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </span>
            </span>
          </div>
          <div
            aria-label="Date"
            aria-sort="descending"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 120px",
                "flex": "0 1 120px",
                "msFlex": "0 1 120px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Date"
            >
              Date
            </span>
            <svg
              className="ReactVirtualized__Table__sortableHeaderIcon ReactVirtualized__Table__sortableHeaderIcon--DESC"
              height={18}
              viewBox="0 0 24 24"
              width={18}
            >
              <path
                d="M7 10l5 5 5-5z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "1 1 200px",
                "flex": "1 1 200px",
                "msFlex": "1 1 200px",
              }
            }
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Description"
            >
              Description
            </span>
          </div>
          <div
            aria-label="Category"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 200px",
                "flex": "0 1 200px",
                "msFlex": "0 1 200px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Category"
            >
              Category
            </span>
          </div>
          <div
            aria-label="In"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 130px",
                "flex": "0 1 130px",
                "msFlex": "0 1 130px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="In"
            >
              In
            </span>
          </div>
          <div
            aria-label="Out"
            aria-sort="none"
            className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 130px",
                "flex": "0 1 130px",
                "msFlex": "0 1 130px",
              }
            }
            tabIndex={0}
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title="Out"
            >
              Out
            </span>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 40px",
                "flex": "0 1 40px",
                "msFlex": "0 1 40px",
              }
            }
          >
            <span
              className="ReactVirtualized__Table__headerTruncatedText"
              title={null}
            />
          </div>
        </div>
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
          onScroll={[Function]}
          role="rowgroup"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": -25,
              "overflowX": "hidden",
              "overflowY": "auto",
              "position": "relative",
              "width": 0,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        />
      </div>
    </div>
  </div>
</div>
`;
